sympy/physics/control/lti.py

```diff
@@ -697,16 +713,54 @@ def to_expr(self):
             return Pow(self.den, -1, evaluate=False)
 
 
-class Series(Basic):
-    """
-    A class for representing product of transfer functions or transfer functions in a
-    series configuration.
+def _mat_mul_compatible(*args):
+    """To check whether shapes are compatible for matrix mul."""
+    return all(
+        arg.num_outputs == args[i].num_inputs
+        for i, arg in enumerate(args, start=1)
+        if i < len(args)
+    )
```

Comment:
This use of enumerate is probably more confusing then just using an integer index like:
```
return all(args[i].num_outputs == args[i+1].num_inputs for i in range(len(args)-1))
```
Otherwise `zip` is the tool to use rather than enumerate:
```
pairs = zip(args[:-1], args[1:])
return all(a.num_outputs == b.num_inputs for a, b in pairs)
```